### PR TITLE
Edit event

### PR DESCRIPTION
**Pull request by PhilipAB**
**What this PR does / why we need it**:
Enable functionality to edit provisioned (currently running) events. 
Provide a route to delete events.

**Which issue(s) this PR fixes**:
Backend implementation for hobbyfarm/hobbyfarm#75
Fixes hobbyfarm/hobbyfarm#76

### CODE REVIEW

File: pkg/controllers/scheduledevent/scheduledeventcontroller.go [hunk 1]
```diff
@@ -353,6 +363,9 @@ func (s ScheduledEventController) provisionScheduledEvent(templates *hfv1.Virtua
 					UID:        se.UID,
 				},
 			},
+			Labels: map[string]string{
+				"scheduledevent": se.Name,
```
Comment (ebauman) [inline pkg/controllers/scheduledevent/scheduledeventcontroller.go:367]: let's namespace this. `scheduledevent.hobbyfarm.io` please

File: pkg/scheduledeventserver/scheduledevent.go [hunk 2]
```diff
@@ -267,7 +269,7 @@ func (s ScheduledEventServer) CreateFunc(w http.ResponseWriter, r *http.Request)
 	scheduledEvent, err = s.hfClientSet.HobbyfarmV1().ScheduledEvents().Create(scheduledEvent)
 	if err != nil {
 		glog.Errorf("error creating scheduled event %v", err)
-		util.ReturnHTTPMessage(w, r, 500, "internalerror", "error creating scehduled event")
+		util.ReturnHTTPMessage(w, r, 500, "internalerror", "error creating scheduled event")
```
Comment (ebauman) [inline pkg/scheduledeventserver/scheduledevent.go:272]: 🤦‍♂️ thanks

**Comment by ebauman** [comment]
@jggoebel @PhilipAB Thank you for the work on this. 

 Currently, the editing and deleting of scheduled events only clears up `virtualmachinesets`. It will also need to clear up `virtualmachines` as scheduled events can run in "on demand" mode which does not create a vmset. Gargantua instead creates `virtualmachines` on-demand as users request a new session.

If you can rework this to also take care of VMs I think we'll be better off.

**Comment by PhilipAB** [comment]
Hi @ebauman, thanks for your quick reply! Before we started, we were looking for what happens when an event is finished. Here again, only the VMSets are deleted: https://github.com/hobbyfarm/gargantua/blob/f8b587cda738414fe83af87255c06788d1dd8cc9/pkg/controllers/scheduledevent/scheduledeventcontroller.go#L146-L188 Shouldn't we then delete the VMs in that case too? 

As far as I understand, according to the current implementation in this pull request, once a scheduled event is deleted, a user cannot request new VMs (since the access code and DBCs are deleted), even if they are on demand. 

However, as you correctly pointed out, a user can still continue to work on existing VMs until the session expires.
Assuming this is an unwanted behavior, we should of course delete them. I/we will take care of that next week.

File: pkg/controllers/scheduledevent/scheduledeventcontroller.go (hunk 1 above)
Comment (jggoebel) [reply pkg/controllers/scheduledevent/scheduledeventcontroller.go]: Good point!

**Comment by jggoebel** [comment]
Hey @ebauman,
we figured out it might the best solution to just invalidate the sessions for finished or deleted events, instead of just deleting the VMs.
This has the same result as clicking on the finish button inside the scenario, and will result in a deletion of the VMs.

The user will then get a message displayed after the next keepalive call (Could we decrease the time for keepalives calls to 15s or something like that?).

**Comment by ebauman** [comment]
Yes I think that's a better approach. I will pull and test this, and I like where it's at so far. Thank you

**Pull request merged**